app/src/main/java/com/github/bgabriel998/softwaredevproject/Button1Activity.java

```diff
@@ -136,12 +151,141 @@ protected void onResume() {
+            params.bottomToBottom = R.id.cameraLayout;
+            params.endToEnd = -1;
+            params.startToStart = -1;
+            params.rightMargin = 10;
+
+        } else if (newConfig.orientation == Configuration.ORIENTATION_PORTRAIT) {
+            params.rightToRight = -1;
+            params.topToTop = -1;
+            params.endToEnd = R.id.cameraLayout;
+            params.startToStart = R.id.cameraLayout;
+            params.bottomToBottom = R.id.compass;
+            params.rightMargin = 0;
+        }
+
+        //Change the constraints of the button
+        takePictureButton.setLayoutParams(params);
+    }
+
+    /**
+     * Callback for the takePicture ImageButton takes two pictures, one of the camera and one with the UI
+     *
+     * @param view ImageButton
+     * @throws IOException if the bitmap could not be stored
+     */
+    public void takePictureListener(View view) throws IOException {
+        //Take a picture with the camera without the UI
+        cameraPreview.takePicture();
+        //Create a bitmap of the camera preview
+        Bitmap cameraBitmap = cameraPreview.getBitmap();
+        //Create a bitmap of the compass-view
+        Bitmap compassBitmap = compassView.getBitmap();
+        //Combine the two bitmaps
+        Bitmap bitmap = overlay(cameraBitmap, compassBitmap);
+        //Store the bitmap on the user device
+        storeBitmap(bitmap);
+    }
+
+    /**
+     * Combines two bitmaps into one
+     *
+     * @param base    first bitmap
+     * @param overlay second bitmap that is drawn on first bitmap
+     * @return A bitmap that combine the two bitmaps
+     */
+    private Bitmap overlay(Bitmap base, Bitmap overlay) {
+        Bitmap bitmap = Bitmap.createBitmap(base.getWidth(), base.getHeight(), base.getConfig());
+        Canvas canvas = new Canvas(bitmap);
+        canvas.drawBitmap(base, new Matrix(), null);
+        canvas.drawBitmap(overlay, new Matrix(), null);
+        return bitmap;
+    }
+
+    /**
+     * Stores the bitmap on the device.
+     *
+     * @param bitmap Bitmap that is to be stored
+     * @throws IOException thrown when bitmap could not be stores
+     */
+    private void storeBitmap(Bitmap bitmap) throws IOException {
+        //Create the file
+        String FILENAME = "yyyy-MM-dd-HH-mm-ss-SSS";
+        String PHOTO_EXTENSION = ".jpg";
+        File screenshotFile = createFile(this, FILENAME, PHOTO_EXTENSION);
+        FileOutputStream outputStream = new FileOutputStream(screenshotFile);
+        int quality = 100;
+        bitmap.compress(Bitmap.CompressFormat.JPEG, quality, outputStream);
+        outputStream.flush();
+        outputStream.close();
+    }
+
+    /**
+     * Creates a file which will be used to store the bitmaps
+     *
+     * @param format    Format of file name
+     * @param extension File extension
+     * @return A file with the
+     */
+    static File createFile(Context context, String format, String extension) {
```

Comment:
This comment goes for more than just this line of code. But could it be an idea to create a separate static class to handle the "createFile, get directory, FILE_NAME, PHOTO_EXTENSION parts. 
The reason why I am asking is that both Button1Activity and the CameraPreview are currently using a few of the same things. Aswell as CameraPreview is using Button1Activitys createFile function. 
I can imagine that the Gallery probably will want to use at least some similar functions to reach the directory. 
Therefore it might be nice to have it all on the same place if possible. What do you think?

Reply:
Yes, I also think that it would be better to create a new class to manage the file creation/storage. However, I think that I will create a new task for the next sprint where I will handle this since I'm already way over the planned 8 hours.